session_0/word_processing.rb

```diff
@@ -34,11 +47,32 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+	l=word_1.size > word_2.size ? word_2.size : word_1.size
+	max=0
+	for i in 1..l
```

Comment:
What are you iterating over in the `for` loop?

Reply:
`l` stores the length of the shorter word. In each iteration we find the substrings of length `i` and compare them. If they are equal we store their length in `max` and if they are not equal we come out of the `for` loop.

Reply:
Right, so you mean the following:

```ruby
similarity_score('applesauce', 'applecake')
- Compare 'a' with 'a'
- Compare 'ap' with 'ap'
- Compare 'app' with 'app'
- Compare 'appl' with 'appl'
- Compare 'apple' with 'apple'
- Compare 'apples' with 'applec'
```

But in the third step - I don't need to compare the complete substrings at all. 

I know that substrings of length 2 are equal and all I need to check if that the third character matches as well i.e. I can compare `word_1[2]` with `word_2[2]` and be done